python/ray/tests/test_actor_failures.py

```diff
@@ -1248,5 +1248,271 @@ def get_pid(self):
     assert ray.get(refs) == [3, 4, 5]
 
 
+def test_actor_cleanup_behavior(ray_start_regular):
+    """Test actor cleanup behavior: __ray_shutdown__ method is called."""
+    import atexit
+    import tempfile
+    import os
+
+    cleanup_file = tempfile.NamedTemporaryFile(delete=False)
+    cleanup_file.close()
+
+    atexit_file = tempfile.NamedTemporaryFile(delete=False)
+    atexit_file.close()
+
+    @ray.remote
+    class CleanupActor:
+        def __init__(self):
+            self.cleanup_file = cleanup_file.name
+            self.atexit_file = atexit_file.name
+            atexit.register(self._atexit_handler)
+
+        def __ray_shutdown__(self):
+            # Write to file to prove cleanup was called
+            with open(self.cleanup_file, "w") as f:
+                f.write("ray_cleanup_called")
+
+        def _atexit_handler(self):
+            # Write to file to prove atexit was called (should NOT happen)
+            with open(self.atexit_file, "w") as f:
+                f.write("atexit_called")
+
+        def get_ready(self):
+            return "ready"
+
+    try:
+        # Test normal actor destruction
+        actor = CleanupActor.remote()
+        ray.get(actor.get_ready.remote())
+        del actor
+
+        # Wait for cleanup
+        time.sleep(1)
```

Comment:
please no sleeps in tests... you can `wait_for_condition` for the assertion just below